Makefile

```diff
@@ -22,6 +23,32 @@ tests: lib mbedtls_test
 mbedtls_test:
 	$(MAKE) -C tests mbedtls_test
 
+library/%:
+	$(MAKE) -C library $*
+programs/%:
+	$(MAKE) -C programs $*
+tests/%:
+	$(MAKE) -C tests $*
+
+.PHONY: generated_files
+generated_files: library/generated_files
+generated_files: programs/generated_files
+generated_files: tests/generated_files
+generated_files: visualc_files
+
+.PHONY: visualc_files
+VISUALC_FILES = visualc/VS2010/mbedTLS.sln visualc/VS2010/mbedTLS.vcxproj
+# TODO: $(app).vcxproj for each $(app) in programs/
+visualc_files: $(VISUALC_FILES)
+$(VISUALC_FILES): scripts/generate_visualc_files.pl
+$(VISUALC_FILES): scripts/data_files/vs2010-app-template.vcxproj
+$(VISUALC_FILES): scripts/data_files/vs2010-main-template.vcxproj
+$(VISUALC_FILES): scripts/data_files/vs2010-sln-template.sln
+# TODO: also the list of .c and .h source files, but not their content
```

Comment:
Note: do we really need a Gnu Makefile target for generating Visual C files? What problem does it solve, who's going to use it? I don't have problem with having one, except apparently we can't easily have proper dependencies, which are most of the point of `make` as opposed to a script, so why not skip this target and just let people invoke the script directly?